database/migrations/2020_01_13_000001_create_order_products_table.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateOrderProductsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('order_product', function (Blueprint $table) {
+            $table->bigIncrements('id');
+            $table->unsignedBigInteger('order_id')->index();
+            $table->foreign('order_id')->references('id')->on('orders');
+            $table->unsignedBigInteger('product_id')->unsigned()->index();
```

Comment:
You don't need the `->unsigned()->index();` 

`unsignedBigInteger` already applies `unsigned` and `index`

Reply:
Are you sure that it applies `index`, too?

Reply:
you know .. im not sure .. i thought i was sure but after digging into the source i can't see where it does.